src/QueryBuilder/ResultViewer/utils.test.js

```diff
@@ -37,3 +37,35 @@ describe('formatValueByDataType returns correct value', () => {
     }
   });
 });
+
+describe('findLabelByValue', () => {
+  const mockOptions = {
+    options: [
+      { value: 'opt_1', label: 'Option 1' },
+      { value: 'opt_2', label: 'Option 2' },
+      { value: '_custom_field_123', label: 'Custom Field 123' },
+    ],
+  };
+
+  it('should return the value if it does not start with _custom_field or opt_', () => {
+    expect(findLabelByValue(mockOptions, 'random_value')).toBe('random_value');
+  });
+
+  it('should return the value if it is an array', () => {
+    expect(findLabelByValue(mockOptions, ['opt_1', 'opt_2'])).toEqual(['opt_1', 'opt_2']);
+  });
+
+  it('should return the label if the value exists in options', () => {
+    expect(findLabelByValue(mockOptions, 'opt_1')).toBe('Option 1');
+    expect(findLabelByValue(mockOptions, '_custom_field_123')).toBe('Custom Field 123');
+  });
+
+  it('should return undefined if the value is not found in options', () => {
+    expect(findLabelByValue(mockOptions, 'opt_999')).toBeUndefined();
+  });
```

Comment:
"should return undefined if the value is a custom field without a match in options"

Alternatively, nest all the `opt_` and `_custom_fields` tests inside their own `describe('custom fields', () => {})` block. As-is, the test description doesn't explain how this test is different than the first one, even though they give different results.